pkg/webhook/admission/pod/metrics_aggregate_injector.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021 The KServe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"github.com/kserve/kserve/pkg/constants"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	defaultPrometheusPort          = "8080"
+	PrometheusPortAnnotationKey    = "prometheus.io/port"
+	PrometheusPathAnnotationKey    = "prometheus.io/path"
+	PrometheusPathAnnotationValue  = "/metrics"
+	QueueProxyAggregateMetricsPort = "9088"
+)
+
+// InjectMetricsAggregator looks for the annotations to enable aggregate kserve-container and queue-proxy metrics and
+// if specified, sets port-related EnvVars in queue-proxy and the aggregate prometheus annotation.
+func InjectMetricsAggregator(pod *v1.Pod) error {
+	for i, container := range pod.Spec.Containers {
+		if container.Name == "queue-proxy" {
+			if enableMetricAgg, ok := pod.ObjectMeta.Annotations[constants.EnableMetricAggregation]; ok && enableMetricAgg == "true" {
```

Comment:
see https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-metrics-merging

Reply:
the prometheus docs link to the prometheus annotation, do you mean that we should add that one to configmap?

Reply:
right now default is false (or no annotation set), are you suggesting we change it to true? the only issue with that I think is the custom queue-proxy image. maybe we want to leave default as not present / false for now.

Reply:
We should add to the configmap but still default to false, when user configures to be true then we enable the metrics merging in the mutator and set the env on queue proxy. Without adding to configmap user needs to add the `EnableMetricAggregation` annotation to every inference service yaml if we want to turn on this feature across board right?

When configmap is set to false, the if condition can be still kept to override the configmap default to enable the metrics merging if you want to turn on selective inference service.

Reply:
Or *it won't set the aggregation env vars/ annotation

Reply:
after discussing online: in the next PR this will be set in config map at a global level to give infra engineers the ability to set a global default for their services. since it doesn't actually scrape metrics until the next PR, it will be added then.